### PR TITLE
WIP: Support for building Rust Lambda functions

### PR DESCRIPTION
**Pull request by sanathkr**
Built by @sapessi originally. The builder works fine as is, but needs PR review and some minor refactors before it can be merged.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_lambda_builders/workflows/rust_cargo/actions.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+from aws_lambda_builders.actions import BaseAction, Purpose, ActionFailedError
+from .cargo import CargoParser, CargoFileNotFoundError, CargoParsingError, CargoValidationError, ExecutableNotFound
+
+LOG = logging.getLogger(__name__)
+TARGET_PLATFORM = "x86_64-unknown-linux-musl"
+
+class CargoValidator(BaseAction):
+    """
+    Validates that Cargo.toml is configured correclty to build a Lambda application
```
Comment (Phrohdoh) [inline aws_lambda_builders/workflows/rust_cargo/actions.py]: s/correclty/correctly

File: aws_lambda_builders/workflows/rust_cargo/cargo.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            Whether the method should print warnings in the LOG object.
+
+        :raise CargoValidationError: If the cargo file was not parsed correctly
+            or it does not represent a valid package.
+        """
+        if not self.cargo:
+            raise CargoValidationError("Cargo file not parsed")
+
+        if "package" not in self.cargo:
+            raise CargoValidationError("Cargo file is not a valid package")
```
Comment (Phrohdoh) [inline aws_lambda_builders/workflows/rust_cargo/cargo.py]: Erroring with something like the following may help users diagnose issues:

> Manifest does not contain required table `package`

File: aws_lambda_builders/workflows/rust_cargo/cargo.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                            "in Cargo.toml file. Consider including a name "
+                            "property and setting its value to bootstrap. This "
+                            "is the executable name AWS Lambda expects."))
+
+
+        if "dependencies" in self.cargo:
+            deps = self.cargo["dependencies"]
+            if "lambda_runtime" not in deps and print_warnings:
+                LOG.warning("""lambda_runtime is not included as a dependency in
+                            Your Cargo.toml file.""")
```
Comment (Phrohdoh) [inline aws_lambda_builders/workflows/rust_cargo/cargo.py]: s/Your/your

**Comment by sapessi** [comment]
Closing this pull request. I'll open a separate one with an updated version of the builder that also addresses the CR feedback.

**Pull request closed without merging**